### PR TITLE
[Docs] Make x86_84 support more explicit

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
In response to #1166

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -2,7 +2,12 @@
...
 
-Note that at this time, BinaryBuilder itself runs on Linux `x86_64` and macOS `x86_64` systems only, with Windows support under active development.  On macOS and Windows, you must have `docker` installed as the backing virtualization engine.  Note that Docker Desktop is the recommended version; if you have Docker Machine installed it may not work correctly or may need additional configuration.
+!!! warning "BinaryBuilder only runs on x86_64 systems"
+    At this time, BinaryBuilder itself runs on Linux `x86_64` and macOS `x86_64` systems only,
+    with Windows support under active development. Explicitly, this means that BinaryBuilder will **not** run on the newer Apple Silicon 'M1' Macs.
```
Comment (giordano) [inline docs/src/index.md:7]: Well, it works with an x86_64 build of Julia on the M1

Comment (jeremiahpslewis) [reply docs/src/index.md:7]: Does it if docker is running natively?

Comment (jeremiahpslewis) [reply docs/src/index.md:7]: (checking myself, ignore above)

Comment (jeremiahpslewis) [reply docs/src/index.md:7]: When I run the following with Julia 1.7 & BinaryBuilder 1.5.1 via Rosetta:
```bash
BINARYBUILDER_AUTOMATIC_APPLE=true julia build_tarballs.jl --debug --verbose aarch64-apple-darwin
```

I get:

```julia
[ Info: Docker base image already exists, skipping import...
 Downloading artifact: PlatformSupport-aarch64-apple-darwin20.v2021.8.10.x86_64-linux-musl.unpacked
[ Info: Unpacking /var/folders/7c/hm96b6vx3r9dc4_s5f6y9m1r0000gn/T/jl_9Ba9Rf-download.gz into /Users/jpslewis/.julia/artifacts/jl_3ZvYJk...
┌ Error: Tree Hash Mismatch!
│   Expected git-tree-sha1:   e09f78a42f59ebfe2f7ea979afc8b3567126de59
│   Calculated git-tree-sha1: dcae944b39f6c9026aa8a6d710084d54b19ac43b
  Downloaded artifact: PlatformSupport-aarch64-apple-darwin20.v2021.8.10.x86_64-linux-musl.unpacked
 Downloading artifact: PlatformSupport-aarch64-apple-darwin20.v2021.8.10.x86_64-linux-musl.unpacked
[ Info: No hash cache found
[ Info: Calculated hash 4256776d422edd94ea420c7dcaf09f2b29df185be8e9e218528e66dd72c80632 for file /var/folders/7c/hm96b6vx3r9dc4_s5f6y9m1r0000gn/T/jl_AFSvpa-download.gz
[ Info: Unpacking /var/folders/7c/hm96b6vx3r9dc4_s5f6y9m1r0000gn/T/jl_AFSvpa-download.gz into /Users/jpslewis/.julia/artifacts/jl_EKJYiR...
┌ Error: Tree Hash Mismatch!
│   Expected git-tree-sha1:   e09f78a42f59ebfe2f7ea979afc8b3567126de59
│   Calculated git-tree-sha1: dcae944b39f6c9026aa8a6d710084d54b19ac43b
  Downloaded artifact: PlatformSupport-aarch64-apple-darwin20.v2021.8.10.x86_64-linux-musl.unpacked
ERROR: LoadError: Unable to automatically install 'PlatformSupport-aarch64-apple-darwin20.v2021.8.10.x86_64-linux-musl.unpacked' from '/Users/jpslewis/.julia/packages/BinaryBuilderBase/khVxP/Artifacts.toml'
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] ensure_artifact_installed(name::String, meta::Dict{String, Any}, artifacts_toml::String; platform::Platform, verbose::Bool, quiet_download::Bool, io::Base.TTY)
    @ Pkg.Artifacts /Applications/Julia-1.7 2.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Pkg/src/Artifacts.jl:441
  [3] ensure_artifact_installed(name::String, artifacts_toml::String; platform::Platform, pkg_uuid::Nothing, verbose::Bool, quiet_download::Bool, io::Base.TTY)
    @ Pkg.Artifacts /Applications/Julia-1.7 2.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Pkg/src/Artifacts.jl:404
  [4] mount(cs::BinaryBuilderBase.CompilerShard, build_prefix::String; verbose::Bool)
    @ BinaryBuilderBase ~/.julia/packages/BinaryBuilderBase/khVxP/src/Rootfs.jl:258
  [5] #41
```

`x86_64-linux-gnu` builds work without a problem. For `x86_64-apple-darwin` I get the same error as above. Does your toolchain run through?

Comment (giordano) [reply docs/src/index.md:7]: What version of Julia is that?

Comment (jeremiahpslewis) [reply docs/src/index.md:7]: Julia 1.7

File: src/AutoBuild.jl [hunk 2]
```diff
@@ -161,6 +161,10 @@ function build_tarballs(ARGS, src_name, src_version, sources, script,
         error("Package name \"$(src_name)\" is not a valid identifier")
     end
 
+    if Sys.ARCH != :x86_64
```
Comment (giordano) [inline src/AutoBuild.jl:164]: I'm somewhat confident we can make it work on any platform which can run an x86_64 docker image with relatively little effort, so this error message isn't much useful. I just don't have much need to go through this all the things that need to be changed

File: docs/src/index.md (hunk 1 above)
Comment (giordano) [reply docs/src/index.md:7]: 1.7.0 has a known bug with computation of tree hashes

File: src/AutoBuild.jl (hunk 2 above)
Comment (jeremiahpslewis) [reply src/AutoBuild.jl:164]: What about if I change the error to something like: 'not currently supported', or switch it to a warning 'needs further development'? Right now it just errors out with a nonspecific failure; the target audience for this change would be 'new person to Yggdrasil, tries building locally with recent Mac and gets a crash which they can't explain / immediately debug'

File: docs/src/index.md (hunk 1 above)
Comment (jeremiahpslewis) [reply docs/src/index.md:7]: aha, ok, let me try 1.7.1

Comment (jeremiahpslewis) [reply docs/src/index.md]: 1.7.0 was the issue.

**Comment by jeremiahpslewis** [comment]
- https://github.com/JuliaPackaging/Yggdrasil/pull/4385
- https://github.com/JuliaPackaging/Yggdrasil/pull/4384
- https://github.com/JuliaPackaging/Yggdrasil/pull/4383

Plus maybe a couple of other fixes should make this PR unnecessary. Closing.

**Pull request closed without merging**